src/NHibernate/Engine/ForeignKeys.cs

```diff
@@ -179,9 +179,16 @@ public static bool IsNotTransientSlow(string entityName, object entity, ISession
 			}
 
 			// let the interceptor inspect the instance to decide
+			if (session.Interceptor.IsTransient(entity) is bool transient)
```

Comment:
Surely it cannot have a significant performance impact in the middle of this processing, but still, using a cast here feels quite hackish to me. Better use a local var and check `HasValue`.
(I have checked the performance anyway : https://dotnetfiddle.net/6wcSBP.)

```suggestion
			var transient = session.Interceptor.IsTransient(entity);
			if (transient.HasValue)
```

Reply:
Done. Hm.. Thanks for benchmarks. I was hoping that compiler would  be smart enough to make it equivalent for var